service/internal/proctelemetry/process_telemetry.go

```diff
@@ -127,6 +155,84 @@ func RegisterProcessMetrics(registry *metric.Registry, ballastSizeBytes uint64)
 	return nil
 }
 
+func (pm *processMetrics) recordWithOtel(meter otelmetric.Meter) error {
+	var err error
+
+	pm.otelProcessUptime, err = meter.Float64ObservableCounter(
+		"process_uptime",
+		instrument.WithDescription("Uptime of the process"),
+		instrument.WithUnit(unit.Unit("s")),
+		instrument.WithFloat64Callback(func(_ context.Context, o instrument.Float64Observer) error {
+			o.Observe(pm.updateProcessUptime())
+			return nil
+		}))
+	if err != nil {
+		return err
+	}
```

Comment:
Instead of checking and returning after first error, I think will be easier to read and probably a bit more helpful to aggregate all errors and return the result:

```suggestion
	var errs, err error

	pm.otelProcessUptime, err = meter.Float64ObservableCounter(
		"process_uptime",
		instrument.WithDescription("Uptime of the process"),
		instrument.WithUnit(unit.Unit("s")),
		instrument.WithFloat64Callback(func(_ context.Context, o instrument.Float64Observer) error {
			o.Observe(pm.updateProcessUptime())
			return nil
		}))
	errs = multierr.Append(errs, err)
```